src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -670,6 +670,42 @@ void QgsWmsCapabilities::parseCapability( QDomElement const &e, QgsWmsCapability
     }
   }
 
+  // Layers in a WMS-C Capabilities TileSet do not have layer title or abstract
+  // still those could be present in a Layer list.
+  if ( !mTileLayersSupported.isEmpty() )
+  {
+    QHash<QString, QString> titles;
+    QHash<QString, QString> abstracts;
+
+    // Build layer identifier - title|abstract mapping
+    for ( const QgsWmsLayerProperty &layer : qgis::as_const( mLayersSupported ) )
+    {
```

Comment:
I think that we could avoid the double check for the layer's name by doing something like this in the first place:

``` c++
for ( const QgsWmsLayerProperty &layer : qgis::as_const( mLayersSupported ) )
{
    if ( layer.name.isEmpty() )
        continue;

    ...
}
```

What do you think?

Reply:
You are right. Done.